acceptance/tests/language/sensitive_data_type.rb

```diff
@@ -0,0 +1,86 @@
+test_name 'C98120, C98077: Sensitive Data is redacted on CLI, logs, reports' do
+require 'puppet/acceptance/environment_utils'
+extend Puppet::Acceptance::EnvironmentUtils
+
+  app_type        = File.basename(__FILE__, '.*')
+  tmp_environment = mk_tmp_environment_with_teardown(master, app_type)
+  fq_tmp_environmentpath  = "#{environmentpath}/#{tmp_environment}"
+
+  # FIXME:
+  #   expand this to other resource types
+  # first attempts at a reasonable table driven test.  needs API work
+  # FIXME: add context messaging to each instance
+  redacted = 'Sensitive \[value redacted\]'
+  test_resources = [
+    {:type => 'notify', :parameters => {:namevar => "1:${Sensitive.new('sekrit1')}"}, :refute => {:match => 'sekrit1'}, :assert => {:match => "1:#{redacted}"}},
+    {:type => 'notify', :parameters => {:namevar => "2:${Sensitive.new($meh2)}"}, :pre_code => '$meh2="sekrit2"', :refute => {:match => 'sekrit2'}, :assert => {:match => "2:#{redacted}"}},
+    {:type => 'notify', :parameters => {:namevar => "3:meh", :message => '"3:${Sensitive.new(\'sekrit3\')}"'}, :refute => {:match => 'sekrit3'}, :assert => {:match => "3:#{redacted}"}},
```

Comment:
agree completely.  i have a file resource i added recently.  the params should be auto-populated into puppet resource code.